src/components/calculator/calculator.jsx

```diff
@@ -2,15 +2,36 @@
 import React, { PureComponent } from 'react';
 import Keyboard from '../keyboard';
 import Container, { ValueDisplay } from './styles';
+import calculate from '../../logic/calculate';
 
 class Calculator extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null || 0,
+      next: null,
+      operation: null,
+    };
+    this.updateValue = this.updateValue.bind(this);
+  }
+
+  updateValue(value) {
+    const newvalue = calculate(this.state, value);
+    this.setState((state) => ({ ...state, ...newvalue }));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <Container>
         <ValueDisplay>
-          <p>0</p>
+          <p>{total}</p>
+          {' '}
+          <p>{operation}</p>
+          {' '}
+          <p>{next}</p>
         </ValueDisplay>
```

Comment:
- [ ] It shows the same value when I perform divide by zero operation (i.e. 9 / 0 = 9 / 0), instead, you can display the `error!` message in the display so that your app gives a better UI.